tests/build-container.sh

```diff
@@ -6,11 +6,49 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 DOCKERFOLDER=$DIR/dockerfile
 REPOFOLDER=$DIR/..
 
-# change our directory sot hat the git arcive command works as expected
-pushd $REPOFOLDER
+# By default we want to do a clean build, but for faster development `USE_LOCAL_ARTIFACTS=1` can
+# be set in which case binaries that reuse local artifacts will be placed into the docker image
+if [[ "${USE_LOCAL_ARTIFACTS:-0}" -eq "0" ]]; then
+    # change our directory so that the git archive command works as expected
+    pushd $REPOFOLDER
+
+    #docker system prune -a -f
+    # Build base container
+    #git archive --format=tar.gz -o $DOCKERFOLDER/gravity.tar.gz --prefix=gravity/ HEAD
+
+    git archive --format=tar.gz -o $DOCKERFOLDER/gravity.tar.gz --prefix=gravity/ HEAD
+else
+    # getting the `test-runner` binary
+    pushd $REPOFOLDER/orchestrator/test_runner && PATH=$PATH:$HOME/.cargo/bin cargo build --all --release
+    # getting the `gravity` binary. `GOBIN` is set so that it is placed under `/dockerfile`.
+    # We check for its existence and rename it to prevent confusion.
+    # This will be moved to its normal place by the `Dockerfile`.
+    pushd $REPOFOLDER/module/ &&
+        PATH=$PATH:/usr/local/go/bin GOPROXY=https://proxy.golang.org make &&
+        PATH=$PATH:/usr/local/go/bin GOBIN=$REPOFOLDER/tests/dockerfile make install &&
+        mv $REPOFOLDER/tests/dockerfile/gravity $REPOFOLDER/tests/dockerfile/go_gravity_bin
+
+    # TODO I would like to also use the local npm artifacts,
+    # but it seems to require more than just copying the binaries
+    # get the ts-node and typescript binaries.
+    #pushd $REPOFOLDER/solidity/ &&
```

Comment:
What you can do here is to copy "node_modules" at least, it would improve the speed as well

Reply:
I tried updating my npm version and broke `npm install` locally (so now I can only build it in the container), so I'm deferring using the local npm artifacts at least until leviathan finishes his npm fixes and I have the time to fix it.

Reply:
We can add this improvement later, so no we go without it, and then when all fixes are done with it :-) Anyway, the update of the artifact should affect the build.